ForumProject/startups/tests.py

```diff
@@ -1,3 +1,160 @@
+from rest_framework_simplejwt.tokens import RefreshToken
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
 from django.test import TestCase
+from startups.models import Startup
+from users.models import CustomUser
+import random, string
 
-# Create your tests here.
+class StartupCreationTestCase(TestCase):
+    
+    def setUp(self):
+        '''
+        Setting up pre-requisites for testing 
+        '''
+        # Create a CustomUser for authentication
+        self.user = CustomUser.objects.create_user(
+            email='dummy@user.com',
+            first_name='John',
+            last_name='Doe',
+            phone_number='+3801234567',
+            password='password',
+            is_active=True
+        )
+        
+        # Get JWT token for the user
+        refresh = RefreshToken.for_user(self.user)
+        self.token = str(refresh.access_token)
+        
+        # Set up the test client and authenticate with the token
+        self.client = APIClient()
+        self.client.credentials(HTTP_AUTHORIZATION='Bearer ' + self.token)
+        
+        # Sample data for testing
+        self.startup_data = {
+                            'startup_name': 'Django Dribblers',
+                            'startup_industry': 'IT',
+                            'startup_phone': '+3801234567',
+                            'startup_country': 'UA',
+                            'startup_city': 'Lviv',
+                            'startup_address': 'I Franka 123'
+                        }
+
+
+    def test_ok_create_startup(self):
+        '''
+        Tested creation of one Startup
+        '''
+        response = self.client.post(reverse('startups:startup-list'), self.startup_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Startup.objects.count(), 1)
+        self.assertEqual(Startup.objects.first().startup_name, 'Django Dribblers')
+
+
+    def test_ok_create_two_startups(self):
+        '''
+        Tested creation of two Startups
+        '''
+        # Create first Startup
+        response = self.client.post(reverse('startups:startup-list'), self.startup_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Startup.objects.count(), 1)
+        
+        # Create second Startup
+        self.startup_data['startup_name'] = 'Python catchers'
+        response = self.client.post(reverse('startups:startup-list'), self.startup_data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(Startup.objects.count(), 2)
+
+        self.assertEqual(Startup.objects.filter(startup_name='Django Dribblers').exists(), True)
+        self.assertEqual(Startup.objects.filter(startup_name='Python catchers').exists(), True)
```

Comment:
Consider using Django’s `@transaction.atomic` decorator for tests that involve multiple database operations which should be rolled back in case of an error during the test
https://docs.djangoproject.com/en/5.0/topics/db/transactions/

Reply:
Done